contracts/DrawAndPrizeDistributionTimelock.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.6;
+import "@pooltogether/v4-core/contracts/interfaces/IDrawBeacon.sol";
+import "@pooltogether/v4-core/contracts/interfaces/IDrawBuffer.sol";
+import "@pooltogether/owner-manager-contracts/contracts/Manageable.sol";
+import "./interfaces/IDrawCalculatorTimelock.sol";
+
+interface IPrizeDistributionFactory {
+  function pushPrizeDistribution(uint32 _drawId, uint256 _totalNetworkTicketSupply) external;
+}
+
+/**
+  * @title  PoolTogether V4 IPrizeDistributionTimelock 
+  * @author PoolTogether Inc Team
+  * @notice The IPrizeDistributionTimelock smart contract interface...
+*/
+interface IDrawAndPrizeDistributionTimelock  {
+  function push(IDrawBeacon.Draw memory draw, uint256 totalNetworkTicketSupply) external;
+}
+
+/**
+  * @title  PoolTogether V4 PrizeDistributionTimelock
+  * @author PoolTogether Inc Team
+  * @notice The PrizeDistributionTimelock smart contract...
```

Comment:
Add a description of what this contract does.

Reply:
👍